### PR TITLE
NIFI-2444 NIFI-2445 fixed PublishKafka

### PR DESCRIPTION
**Pull request by olegz**
- fixed the logging issue NIFI-2444 by ensuring the ProcessLog is added to KafkaPublisher
- fixed KafkaPublisher's isAllAcked operation to ensure that it properly reports that the flow file has failed.
- added additional test

### CODE REVIEW

**Comment by JPercivall** [comment]
Reviewing

**Comment by brosander** [comment]
Checkstyle error
```
[INFO] --- maven-checkstyle-plugin:2.15:check (check-style) @ nifi-kafka-processors ---
[WARNING] src/main/java/org/apache/nifi/processors/kafka/KafkaPublisher.java[36:8] (imports) UnusedImports: Unused import - org.apache.nifi.logging.ComponentLog.
```

File: nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/processors/kafka/KafkaPublisher.java [hunk 1]
```diff
@@ -233,12 +230,10 @@ void setProcessLog(ProcessorLog processLog) {
      */
     private void warnOrError(String message, Exception e) {
         if (e == null) {
-            logger.warn(message);
             if (this.processLog != null) {
```
Comment (markap14) [inline nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/processors/kafka/KafkaPublisher.java]: This feels a little off to me, @olegz - why not just pass the ComponentLog in to the constructor? It should always be present.

File: nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-pubsub-processors/src/test/java/org/apache/nifi/processors/kafka/pubsub/StubPublishKafka.java [hunk 2]
```diff
@@ -85,21 +95,31 @@ protected KafkaPublisher buildKafkaResource(ProcessContext context, ProcessSessi
...
-                if ("futurefail".equals(value) && !StubPublishKafka.this.failed) {
-                    StubPublishKafka.this.failed = true;
-                    when(future.get(Mockito.anyLong(), Mockito.any(TimeUnit.class))).thenThrow(ExecutionException.class);
-                }
+
+                Future<RecordMetadata> future = executor.submit(new Callable<RecordMetadata>() {
+                    @Override
+                    public RecordMetadata call() throws Exception {
+                        if ("futurefail".equals(value) && !StubPublishKafka.this.failed) {
+                            // System.out.println("FAIL");
```
Comment (markap14) [inline nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-pubsub-processors/src/test/java/org/apache/nifi/processors/kafka/pubsub/StubPublishKafka.java]: Can remove this line?

**Comment by olegz** [comment]
@markap14 done

**Comment by olegz** [comment]
@markap14 @brosander done

**Comment by markap14** [comment]
Thanks @olegz. All looks good. +1 merged to 0.x

**Pull request merged**